src/elisp/treemacs-icons.el

```diff
@@ -426,6 +425,221 @@ Necessary since root icons are not rectangular."
     (treemacs-create-icon :file "vsc/r.png"         :extensions ("r"))
     (treemacs-create-icon :file "vsc/reason.png"    :extensions ("re" "rei"))))
 
+(defvar all-the-icons-icon-alist nil)
+(declare-function all-the-icons-octicon "ext:all-the-icons" t t)
+(declare-function all-the-icons-faicon "ext:all-the-icons" t t)
+(declare-function all-the-icons-material "ext:all-the-icons" t t)
+(declare-function all-the-icons-alltheicon "ext:all-the-icons" t t)
+
+(when (require 'all-the-icons nil t)
```

Comment:
Since we require a hard dependency to another package the whole theme should become its own package. All the relevant code should be moved into its own file in src/extra. I'll take care of publishing on melpa once this is merged.

Reply:
I see, I'll move it!